docs/manual-EN/3.build-develop-and-administration/1.build/1.build-source-code.md

```diff
@@ -1,244 +1,200 @@
+For your reference, it is expected to take about 35 minutes in CPU time(less than 4 minutes with `-j16`), given an Intel E5-2697 v3 processor and unit tests are disabled.
 
-```bash
-bash> cd nebula/build
-bash> cmake -DENABLE_TESTING=OFF ..
-bash> make -j$(nproc) && make install -j$(nproc)
-```
+Until now, you might already have built **Nebula Graph** successfully. If so or not, we also provide ways to tweak the building process.
 
-- ENABLE\_ASAN: Add AddressSanitizer when compiling.
+## Ways to Tweak the Building
 
-The default value is OFF. You can set it to ON during cmake with the following command.
+### CMake Arguments/Variables
+We provide several options to make one tweak the building, while some of these are builtins from CMake. These arguments are used at the configure(cmake) stage, like `cmake -DArgument=Value ..`.
 
-```bash
-bash> cd nebula/build
-bash> cmake -DENABLE_ASAN=ON ..
-bash> make -j$(nproc) && make install -j$(nproc)
-```
+#### ENABLE_TESTING
+This option allows to enable or disable the build of unit tests. We suggest to turn it off if you just need the service modules of **Nebula Graph**.
+Optionals are `ON` and `OFF`, and `ON` by default.
 
-#### **Building Completed**
+#### ENABLE_ASAN
+This option enables or disables the ASan building, a.k.a AddressSanitizer, which is a memory error detector. It is meant to be used by **Nebula Graph** developers. This options is disabled by default.
 
-- If no errors are shown
+#### CMAKE_BUILD_TYPE
+There are a few building types supported:
+ * `Debug`, to build with debuginfo but without optimization, which is by default
+ * `Release`, to build with optimization but without debuginfo
+ * `RelWithDebInfo`, to build with optimization AND debuginfo
+ * `MinSizeRel`, to build with optimizations for code size
 
-    ```bash
-    [100%] Built target ....
-    ```
-
-    **Congratulations! Compile successfully...**
-
-- You can see there are four folders in the the installation directory **/usr/local/nebula**: **etc/**, **bin/**, **scripts/**, **share/**.
-
-    ```bash
-    [root@centos7.5 nebula]# ls /usr/local/nebula/
-    bin  etc  scripts  share
-    ```
-
-**Now, you can start Nebula Graph!**
-
-#### Start Nebula Graph Services
-
-After the above installation is completed, you can start the Nebula Graph services. The default directory of Nebula Graph is `/usr/local/nebula`.
+#### CMAKE_INSTALL_PREFIX
+This option is to specify the location where the service modules, scripts, configuration files and tools are installed when `make install` is performed. It is set to `/usr/local/nebula` by default.
 
+#### CMAKE_CXX_COMPILER
+Normally, CMake will figure out and locate an applicable C++ compiler for us. But if your compiler installation is not at the standard location, or if you want to use a different one, you have to specify it explicitly, like:
```

Comment:
Normally, CMake will figure out and locate an applicable C++ compiler for you. But if your compiler installation is not at the default location, or if you want to use a different one, you have to specify it explicitly as follows: